framework/include/base/SubProblem.h

```diff
@@ -368,6 +400,12 @@ class SubProblem : public Problem
   /* This needs to remain <unsigned int> for threading purposes */
   std::vector<unsigned int> _has_active_elemental_moose_variables;
 
+  /// Set of material property ids that determine whether materials get reinited
+  std::vector<std::set<unsigned int>> _active_material_property_ids;
+
+  /// True if _active_material_property_ids is not empty
+  std::vector<unsigned int> _has_active_material_property_ids;
```

Comment:
This should be ``vector<bool>`` and all usages should use ``true`` and ``false``.  It might actually be worth just using ``_active_material_property_ids[thread_id].empty()`` instead and removing this second vector entirely - then we don't have to worry about the two getting out of sync.